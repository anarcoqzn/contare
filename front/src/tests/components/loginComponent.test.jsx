import LoginScreen from '../../components/loginComponent/loginComponent';
import Users from '../fixtures/users';
import { shallowToJson } from 'enzyme-to-json';


it('should render correctly', () => {
    const wrapper = shallow(<LoginScreen />);
    expect(wrapper).toMatchSnapshot();
});

it('Should render correctly passing an user', () => {
    const wrapper = shallow(<LoginScreen user = {Users[1]}/>);
    expect(wrapper).toMatchSnapshot();
});

it('render email input label given the type', () => {
    const wrapper = shallow(<LoginScreen type = 'email' required = 'true'/>);
    const label = wrapper.find('Form.Control');
    expect(label).toHaveLength(0);
});

it('should handle handleSubmit', () => {
    let handleSubmit = jest.fn();
    let history = {push: jest.fn()};
    const wrapper = shallow(<LoginScreen handleSubmit = {handleSubmit} history= {history}/>);
    wrapper.find('Button').prop('submit');
});
