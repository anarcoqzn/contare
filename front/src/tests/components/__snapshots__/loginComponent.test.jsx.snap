// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly passing an user 1`] = `
<div>
  <h3>
    Entrar
  </h3>
  <[object Object]
    as="form"
    inline={false}
    noValidate={true}
    onSubmit={[Function]}
    validated={false}
  >
    <[object Object]
      as="div"
      controlId="formBasicEmail"
    >
      <[object Object]
        column={false}
        srOnly={false}
      >
        Email
      </[object Object]>
      <[object Object]
        as="input"
        placeholder="Digite seu email"
        required={true}
        type="email"
      />
      <[object Object]
        as="div"
        type="invalid"
      >
        Por favor digite o email.
      </[object Object]>
    </[object Object]>
    <[object Object]
      as="div"
      controlId="formBasicPassword"
    >
      <[object Object]
        column={false}
        srOnly={false}
      >
        Senha
      </[object Object]>
      <[object Object]
        as="input"
        placeholder="Digite sua senha"
        required={true}
        type="password"
      />
      <[object Object]
        as="div"
        type="invalid"
      >
        Por favor digite sua senha.
      </[object Object]>
    </[object Object]>
    <[object Object]
      active={false}
      disabled={false}
      type="submit"
      variant="primary"
    >
      Entrar
    </[object Object]>
  </[object Object]>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</div>
`;

exports[`should render correctly 1`] = `
<div>
  <h3>
    Entrar
  </h3>
  <[object Object]
    as="form"
    inline={false}
    noValidate={true}
    onSubmit={[Function]}
    validated={false}
  >
    <[object Object]
      as="div"
      controlId="formBasicEmail"
    >
      <[object Object]
        column={false}
        srOnly={false}
      >
        Email
      </[object Object]>
      <[object Object]
        as="input"
        placeholder="Digite seu email"
        required={true}
        type="email"
      />
      <[object Object]
        as="div"
        type="invalid"
      >
        Por favor digite o email.
      </[object Object]>
    </[object Object]>
    <[object Object]
      as="div"
      controlId="formBasicPassword"
    >
      <[object Object]
        column={false}
        srOnly={false}
      >
        Senha
      </[object Object]>
      <[object Object]
        as="input"
        placeholder="Digite sua senha"
        required={true}
        type="password"
      />
      <[object Object]
        as="div"
        type="invalid"
      >
        Por favor digite sua senha.
      </[object Object]>
    </[object Object]>
    <[object Object]
      active={false}
      disabled={false}
      type="submit"
      variant="primary"
    >
      Entrar
    </[object Object]>
  </[object Object]>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</div>
`;
